### PR TITLE
Use datalib

### PR DESCRIPTION
**Pull request by domoritz**

### CODE REVIEW

File: src/Encoding.js [hunk 1]
```diff
@@ -194,14 +194,6 @@ var Encoding = module.exports = (function() {
     return sort;
   };
 
-  proto.any = function(f) {
```
Comment (kanitw) [inline src/Encoding.js]: Have you check that these `Encoding.all/any` aren't being used in any dependent projects?

Comment (domoritz) [inline src/Encoding.js]: A little bit. If I notice they are needed, I'll add them to datalib.

Comment (domoritz) [inline src/Encoding.js]: Once you merged this, I'll port polestar and voyager.

File: src/util.js [hunk 2]
```diff
@@ -1,175 +1,13 @@
...
-
-util.isArray = Array.isArray || function(obj) {
-  return toString.call(obj) == '[object Array]';
-};
-
-util.array = function(x) {
-  return x ? (util.isArray(x) ? x : [x]) : [];
-};
-
 util.forEach = function(obj, f, thisArg) {
```
Comment (kanitw) [inline src/util.js]: shouldn't this be included in datalib too?

File: src/vl.js [hunk 3]
```diff
@@ -4,7 +4,10 @@ var globals = require('./globals'),
     util = require('./util'),
     consts = require('./consts');
 
-var vl = util.merge(consts, util);
+var vl = {};  // util.merge(consts, util);
```
Comment (kanitw) [inline src/vl.js]: This will break dependent projects ...

File: src/util.js (hunk 2 above)
Comment (domoritz) [inline src/util.js]: Not really. I think this is a bad implementation and would rather delete it.

File: src/Encoding.js (hunk 1 above)
Comment (kanitw) [inline src/Encoding.js]: compass too.

Comment (domoritz) [inline src/Encoding.js]: I can try.

File: src/vl.js (hunk 3 above)
Comment (domoritz) [inline src/vl.js]: Will fix those. The problem is that if everything is in the same namespace it's really hard to find anything.

File: src/util.js (hunk 2 above)
Comment (kanitw) [inline src/util.js]: Why is this bad?   We don't have a way to call foreach for object and this is it.   
Lodash and underscore provide similar method.

File: src/vl.js (hunk 3 above)
Comment (kanitw) [inline src/vl.js]: I think splitting might be better, but then you should submit a concurrent pr in the other 3 projects for these then.

File: test/data.spec.js [hunk 4]
```diff
@@ -13,11 +14,7 @@ describe('getSchema', function() {
       {a: '', b: null, d: undefined},
       {a: 1, b: 's', d: 'Jan 11, 2014'}
     ];
-    var s = data.getSchema(d);
```
Comment (kanitw) [inline test/data.spec.js]: are these two exactly the same? .... not sure.

File: src/vl.js (hunk 3 above)
Comment (domoritz) [inline src/vl.js]: If you reject this, I do all the work for nothing.

File: src/util.js (hunk 2 above)
Comment (domoritz) [inline src/util.js]: The array version should use `for(var i; i<n;i++)` for performance reasons. Anyway, I'm okay with keeping this here for now but let's not push it into datalib.

File: test/data.spec.js (hunk 4 above)
Comment (domoritz) [inline test/data.spec.js]: Should be. With the sorting below.

File: src/vl.js (hunk 3 above)
Comment (kanitw) [inline src/vl.js]: I'm not rejecting, I am just saying please fix dependencies downstream.

Comment (domoritz) [inline src/vl.js]: Ok

**Comment by kanitw** [comment]
Please merge master into this branch too.  (I added `vl.version` and a few more things earlier today.)

**Comment by domoritz** [comment]
> Please merge master into this branch too. (I added vl.version and a few more things earlier today.)

Done

File: src/util.js (hunk 2 above)
Comment (kanitw) [inline src/util.js]: Array has `foreach`, so it would call the native’s foreach anyway.

Comment (kanitw) [inline src/util.js]: In any case, we would never call this method, if the collection is an array.

File: src/vl.js (hunk 3 above)
Comment (kanitw) [inline src/vl.js]: Thinking more about it.  This is a bad idea. I’ll revert this part — see 536b3d5e26b7070a33d1a6906a71418eb888e2a2

File: src/util.js [hunk 5]
```diff
@@ -180,62 +18,6 @@ util.forEach = function(obj, f, thisArg) {
...
-    return obj.map.call(thisArg, f);
-  } else {
-    var output = [];
-    for (var k in obj) {
-      output.push( f.call(thisArg, obj[k], k, obj));
-    }
-  }
-};
-
-util.any = function(arr, f) {
```
Comment (kanitw) [inline src/util.js]: @domoritz — FYI, you broke compass by removing this ...

Comment (domoritz) [inline src/util.js]: Hmm, I guess, we can either move it into datalib or compass, since it was unused in vl.

**Pull request merged**